interpreter/src/objects.rs

```diff
@@ -486,8 +486,12 @@ impl<'a> Value {
             }
             Expression::And(left, right) => {
                 let left = Value::resolve(left, ctx)?;
-                let right = Value::resolve(right, ctx)?;
-                Value::Bool(left.to_bool() && right.to_bool()).into()
+                if !left.to_bool() {
+                    left.into()
+                } else {
+                    let right = Value::resolve(right, ctx)?;
+                    right.into()
```

Comment:
This may not convert into a bool, the into just turns into a Value. We need to explicitly convert to bool so the result of this expression always returns bool.